docs/HPOS.md

```diff
@@ -0,0 +1,86 @@
+# HPOS
+
+WooCommerce has traditionally stored store orders and related order information (like refunds) as custom WordPress post types or post meta records. This comes with performance issues, and that's why HPOS (High Performance Order Storage) was developed. HPOS is the WooCommerce engine that stores orders in dedicated tables.
+
+HPOS is also referred to as COT (Custom Order Tables) in some parts of the code, that's the early name of the engine.
+
+There are a number of settings that control HPOS operation. Boolean settings are stored using the usual WooCommerce convention: `yes` for enabled ("set to true"), `no` for disabled ("set to false").
+
+Most of the code related to HPOS is in [src/Internal/DataStores/Orders](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/src/Internal/DataStores/Orders).
```

Comment:
Slightly subjective/a matter of style, but I wonder if bullets would work better for the final three sentences?

```suggestion
WooCommerce has traditionally stored store orders and related order information (like refunds) as custom WordPress post types or post meta records. This comes with performance issues, and that's why we developed HPOS (High-Performance Order Storage). It is, quite simply, an engine that stores orders in dedicated tables.

Some high level things you should be aware of:

- Most of the code related to HPOS is contained in [src/Internal/DataStores/Orders](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/src/Internal/DataStores/Orders).
- In some parts of the code, you may find references to COT (Custom Order Tables). That's the earlier name of the engine.
- There are a number of settings that control HPOS operation. Boolean settings are stored using the usual WooCommerce convention: `yes` for enabled ("set to true"), `no` for disabled ("set to false").

```